tuxemon/event/actions/choice_monster.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: GPL-3.0
+# Copyright (c) 2014-2023 William Edwards <shadowapex@gmail.com>, Benjamin Bean <superman2k5@gmail.com>
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from functools import partial
+from typing import final
+
+from tuxemon.event import get_npc
+from tuxemon.event.eventaction import EventAction
+from tuxemon.locale import T, replace_text
+from tuxemon.states.choice.choice_monster import ChoiceMonster
+
+logger = logging.getLogger(__name__)
+
+
+@final
+@dataclass
+class ChoiceMonsterAction(EventAction):
+    """
+    Ask the player to make a choice among monsters.
+
+    Script usage:
+        .. code-block::
+
+            choice_monster <choices>,<variable>
+
+    Script parameters:
+        choices: List of possible choices
+            (monster slugs eg: rockitten:apeoro),
+            separated by a colon ":".
+        variable: Variable to store the result of the choice.
+
+    """
+
+    name = "choice_monster"
+
+    choices: str
+    variable: str
+
+    def start(self) -> None:
+        def set_variable(var_value: str) -> None:
```

Comment:
I feel like inner methods often add unnecessary complexity to the code, what would you think about moving this method outside as a (private) instance method?
`player` would become an argument.
Suggested signature:
```suggestion
        def _set_variable(self, value: str, player: Player) -> None:
```

Reply:
I agree, it makes sense